src/emu/diimage.cpp

```diff
@@ -513,67 +496,72 @@ bool device_image_interface::load_software_region(const char *tag, optional_shar
 // to be loaded
 // ****************************************************************************
 
-void device_image_interface::run_hash(util::core_file &file, void (*partialhash)(util::hash_collection &, const unsigned char *, unsigned long, const char *),
-	util::hash_collection &hashes, const char *types)
+bool device_image_interface::run_hash(util::core_file &file, uint32_t skip_bytes, util::hash_collection &hashes, const char *types)
 {
-	u32 size;
-	std::vector<u8> buf;
-
+	// reset the hash; we want to override existing data
 	hashes.reset();
-	size = (u32) file.size();
 
-	buf.resize(size);
-	memset(&buf[0], 0, size);
+	// figure out the size, and "cap" the skip bytes
+	uint64_t size = file.size();
+	skip_bytes = (uint32_t) std::min((uint64_t) skip_bytes, size);
 
-	// read the file
-	file.seek(0, SEEK_SET);
-	file.read(&buf[0], size);
+	// seek to the beginning
+	file.seek(skip_bytes, SEEK_SET);
+	uint64_t position = skip_bytes;
 
-	if (partialhash)
-		partialhash(hashes, &buf[0], size, types);
-	else
-		hashes.compute(&buf[0], size, types);
+	// keep on reading hashes
+	hashes.begin(types);
+	while(position < size)
```

Comment:
Still missing a space after the flow control keyword, in contrast to the `if` statement below it.

Also, casts are right-associative, so it should be `(uint32_t)whatever` (no space between right-associative operator and argument), or even `uint32_t(whatever)` (function-syntax cast).

And while we're at it, this is in `src/emu`, so it should be using the abbreviates `u8`/`s8`/`u16`/`s16`/`u32`/`s32`/`u64`/`s64` types for consistency with the rest of `src/emu` (same applied to the header).